### PR TITLE
feat: implement the abort signal in `/edit` diff gen

### LINKED ISSUE #1012: [CON-223] Cmd/Ctrl + Shift + Backspace doesn't stop streaming of diff
### Before submitting your bug report

- [X] I believe this is a bug. I'll try to join the [Continue Discord](https://discord.gg/NWtdYexhMs) for questions
- [X] I'm not able to find an [open issue](https://github.com/continuedev/continue/issues?q=is%3Aopen+is%3Aissue) that reports the same bug
- [X] I've seen the [troubleshooting guide](https://continue.dev/docs/troubleshooting) on the Continue Docs

### Relevant environment info

```Markdown
- OS:Mac
- Continue:0.9.91
- IDE:VS Code
```

### Description

When the diff view is briefly going away it is because I am using the cmd+shift+backspace shortcut to cancel the generation, but after a second it comes back because streaming was not successfully cancelled.

https://github.com/continuedev/continue/assets/33237525/8fd80572-8fe9-4330-a563-2d7d59239d94

[This](https://github.com/continuedev/continue/blob/221d020053ad36ca39dd156776dd711a6920f77f/extensions/vscode/src/diff/horizontal.ts#L273) is the function that gets called and is responsible for cancelling, so this is the starting point to debug.

### To reproduce

1. Highlight code
2. Type '/edit fix this' or something similar
3. when it starts generating, press cmd/ctrl+shift+backspace
4. it will briefly go away but streaming continues and the diff view will return

### Log output

_No response_

<sub>[CON-223](https://linear.app/continue/issue/CON-223/cmdctrl-shift-backspace-doesnt-stop-streaming-of-diff)</sub>

### PR DESCRIPTION
**Pull request by Iamshankhadeep**
fixes #1012

### CODE REVIEW

File: gui/src/util/ide.ts [hunk 1]
```diff
@@ -114,6 +114,10 @@ export async function* ideStreamRequest<T extends keyof WebviewProtocol>(
   let returnVal = undefined;
 
   const handler = (event: { data: Message }) => {
+    // console.log("Received message", event.data);
```
Comment (justinmilner1) [inline gui/src/util/ide.ts]: remove

**Review by justinmilner1 — APPROVED** [review]

**Comment by sestinj** [comment]
@justinmilner1 yes this looks perfect! @Iamshankhadeep would you want to adjust to use this solution, or would it be most convenient if I went ahead and just changed this locally and pushed?

**Comment by Iamshankhadeep** [comment]
@sestinj  I have updated the PR, with the new code that @justinmilner1 suggested.

**Comment by justinmilner1** [comment]
I do think the 'shift+ctrl+backspace' shortcut should be added to package.json keybindings pathway, to match the documentation.
I don't think the gui.tsx pathway can provide the cancellation without major refactoring.

**Comment by justinmilner1** [comment]
Hey @Iamshankhadeep - Do you want to wrap this one up?

**Comment by justinmilner1** [comment]
Okay I'm gonna get this closed out

**Comment by sestinj** [comment]
Because of a refactor that happened I ended up having to tweak the way the "abort" message was passed, but the basic solution still held: https://github.com/continuedev/continue/commit/5bf08de42c73265230042aaf186f5bc08e5e4efd

Closing this now, but thank you for figuring out how to do it!

**Pull request closed without merging**